include/flux/op/filter_map.hpp

```diff
@@ -0,0 +1,68 @@
+
+// Copyright (c) 2024 Tristan Brindle (tcbrindle at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef FLUX_OP_FILTER_MAP_HPP_INCLUDED
+#define FLUX_OP_FILTER_MAP_HPP_INCLUDED
+
+#include <flux/core.hpp>
+#include <flux/op/map.hpp>
+
+namespace flux {
+
+    namespace detail {
+
+        struct filter_map_fn {
+            // If dereffing the optional would give us an rvalue reference,
+            // prevent a probable dangling reference by returning by value instead
+            template <typename T>
+            using strip_rvalue_ref_t = std::conditional_t<
+                std::is_rvalue_reference_v<T>, std::remove_reference_t<T>, T>;
+            template <adaptable_sequence Seq, typename Func>
+                requires (std::invocable<Func&, element_t<Seq>> &&
+                          optional_like<std::remove_cvref_t<std::invoke_result_t<Func&, element_t<Seq>>>>)
+            constexpr auto operator()(Seq&& seq, Func func) const
```

Comment:
Formatting: add a blank line between the using declaration and the following declaration for readability

```suggestion
            template <typename T>
            using strip_rvalue_ref_t = std::conditional_t<
                std::is_rvalue_reference_v<T>, std::remove_reference_t<T>, T>;
                
            template <adaptable_sequence Seq, typename Func>
                requires (std::invocable<Func&, element_t<Seq>> &&
                          optional_like<std::remove_cvref_t<std::invoke_result_t<Func&, element_t<Seq>>>>)
            constexpr auto operator()(Seq&& seq, Func func) const
```